drivers/dma/dma_mcux_lpc.c

```diff
@@ -41,11 +44,18 @@ struct call_back {
 	bool busy;
 };
 
+struct dma_otrig {
+	int8_t source_channel;
+	int8_t linked_channel;
```

Comment:
it does not look like there is any reason to keep track of the linked channel, only the source channel, and by having this struct and extra data and lots more conditional checks involving it, the code is getting a lot more complex and hard to understand